docs/migrating_to_9.md

```diff
@@ -36,3 +36,17 @@ try {
   // Handle validation error
 }
 ```
+
+## Errors in middleware functions take priority over `next()` calls
```

Comment:
Wouldnt kareems change of not accepting arguments to `next()` also need to be noted?

Reply:
done 1ab5638ac0d1984f21a95df57a7108d29d4607b5.